### ISSUE TITLE
Pub-Sub support

### ISSUE THREAD

**Issue by tomusdrw** [tomusdrw/rust-web3#37]
Requires #14

**Comment by rphmeier** [comment]
```rust
trait DuplexTransport: Transport {
    type Stream: Stream<Item=rpc::Value, Error=Error>; // drop should unsubscribe.
    type Future: Future<Item=Self::Stream, Error=Error>;

    fn subscribe(request) -> Self::Future;
}

// example method implementation
let stream = self.transport.subscribe(my_subscription_request).and_then(|s| s.map(lazy_decode));
```

How should it work with batch transport? Should you be able to batch regular requests with subscriptions?

Maybe a general stream wrapper for encapsulating subscription ID and dropping logic?

**Comment by tomusdrw** [comment]
Resolved via #105
